docs/WitnessMatchingManifesto.md

```diff
@@ -0,0 +1,363 @@
+# Protocol Witness Matching Manifesto
+
+Author: [Suyash Srijan](https://www.github.com/theblixguy) 
+
+### Jargon
+
+Before we dive into the document, let's clarify some jargon:
+
+- A _protocol requirement_ (or just _requirement_) is a declaration inside a protocol that all conforming types must satisfy.
+- A _protocol witness_ (or just _witness_) is a value or a type that satisfies a protocol requirement.
```

Comment:
Should a protocol witness be described as "a _declaration_" instead of "a _value_"?

```suggestion
- A _protocol witness_ (or just _witness_) is a declaration or a type that satisfies a protocol requirement.
```

One might say that associated type witnesses are technically declarations too: either a type declaration like `struct Foo` or a typealias like `typealias Foo = ...`. But I think it's actually clearer to make a distinction and state "a declaration or a type".